src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,170 @@
-import { useState } from 'react';
+import {
+  FC, FormEvent, useCallback, useState,
+} from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setimgUrl] = useState('');
+  const [imdbUrl, setimdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const [hasTitleError, setTitleError] = useState(false);
+  const [hasDescriptionError, setDescriptionError] = useState(false);
+  const [hasImgUrlError, setImgUrlError] = useState(false);
+  const [hasimdbUrlError, setimdbUrlError] = useState(false);
+  const [hasimdbIDError, setimdbIdError] = useState(false);
```

Comment:
```suggestion
  const [hasImdbUrlError, setHasImdbUrlError] = useState(false);
  const [hasImdbIDError, setHasImdbIdError] = useState(false);
```
Don't forget about the camelCase.
Also, it is a good tone to name set function according to the noun variable.